sources/PTYSession.m

```diff
@@ -3061,29 +3066,26 @@ - (void)setBlend:(float)blendVal {
     [_textview setBlend:blendVal];
 }
 
-- (BOOL)antiIdle
-{
+- (BOOL)antiIdle {
     return _antiIdleTimer ? YES : NO;
 }
 
-- (void)setAntiIdle:(BOOL)set
-{
-    if (set == [self antiIdle]) {
-        return;
+- (void)setAntiIdle:(BOOL)set {
+    [_antiIdleTimer invalidate];
+    [_antiIdleTimer release];
+    _antiIdleTimer = nil;
+    
+    // FIXME: corner case: part 1of2 of a kludgey way to prevent a flood when period is set to 1
+    if (_antiIdlePeriod > 0 && _antiIdlePeriod < 1.1) {
+        _antiIdlePeriod = 1.1;
     }
-
-    if (set) {
-        NSTimeInterval period = MIN(60, [iTermAdvancedSettingsModel antiIdleTimerPeriod]);
-
-        _antiIdleTimer = [[NSTimer scheduledTimerWithTimeInterval:period
+    
+    if (set && _antiIdlePeriod > 0) {
+        _antiIdleTimer = [[NSTimer scheduledTimerWithTimeInterval:_antiIdlePeriod
```

Comment:
I started to do this:
```obj-c
    static const int kMinimumAntiIdlePeriod = 2;
    
    _antiIdlePeriod = MAX(_antiIdlePeriod, kMinimumAntiIdlePeriod);
    
    if (set) {
        _antiIdleTimer = [[NSTimer scheduledTimerWithTimeInterval:_antiIdlePeriod
```

But shouldn't this really be in the ui? So they can't even enter 0 or 1, and it will never say 0 or 1 in the ui yet actually do 2. Not sure if you want that const defined somewhere else in a header or at the top of the file instead of right there in the function either.